src/main/java/duke/commands/SnoozeCommand.java

```diff
@@ -0,0 +1,44 @@
+package duke.commands;
+
+import duke.commons.DukeException;
+import duke.commons.MessageUtil;
+import duke.storage.Storage;
+import duke.tasks.TaskWithDates;
+import duke.ui.Ui;
+
+import java.time.LocalDateTime;
+
+/**
+ * Class representing a command to snooze/reschedule/postpone a task.
+ */
+public class SnoozeCommand extends Command {
+    private int index;
+    private LocalDateTime newDate;
+
+    /**
+     * Creates a new SnoozeCommand with the given index and newDate of the task.
+     *
+     * @param index The index of the task.
+     */
+    public SnoozeCommand(int index, LocalDateTime newDate){
+        this.index = index;
+        this.newDate = newDate;
+    }
+
+    /**
+     * Executes this command on the given task list and user interface.
+     *
+     * @param ui The user interface displaying events on the task list.
+     * @param storage The duke.storage object containing task list.
+     */
+    @Override
+    public void execute(Ui ui, Storage storage) throws DukeException {
+        try {
+            TaskWithDates task =  storage.getTasksWithDate().get(index).updateDate(newDate);
```

Comment:
Task t = storage.getTasks().get(index);
if (t instanceof TaskWithDate) {
 typecast and updates
} else {
//task is todo or something else, so throw DukeExceptions for user to know what when wrong
}